clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -234,23 +245,28 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                 List<TopicPartition> consumerAssignment = assignment.get(consumer);
 
                 int expectedAssignedCount = numMaxCapacityMembers < numExpectedMaxCapacityMembers ? maxQuota : minQuota;
+                int currentAssignedCount = consumerAssignment.size();
                 if (unassignedPartitionsIter.hasNext()) {
                     TopicPartition tp = unassignedPartitionsIter.next();
                     consumerAssignment.add(tp);
-                    unassignedPartitionsIter.remove();
```

Comment:
Don't do the `unassignedPartitions` list element removal as mentioned, it's slow. Instead, we make sure the `unfilledMembers` will be empty in the end inside the while loop. After all, the `unassignedPartitions` size will always >= `unfilledMembers` size.